### ISSUE TITLE
Failed prop type: The prop `i18n` is marked as required ... but its value is `undefined`

### ISSUE THREAD

**Issue by Kernle32DLL** [lingui/js-lingui#291]
**Describe the bug**
While updating to 2.5.0 (2.4.2 works fine), I encountered some warnings on the console while using `withI18n`. In essence, it seems that the `i18n` property is not defined while creating the initial component. The output is as follows:

```
 Warning: Failed prop type: The prop `i18n` is marked as required in `HeadTitle`, but its value is `undefined`.
    in HeadTitle (created by WithI18n)
    in WithI18n (created by PageContent)
    ...
    in I18nProvider (created by App)
    ...
```

Funnily enough, when I make `HeadTitle` a class, and inspect the properties in the constructor, everything is there. So no idea when the prop types are checked, or put differently - why or when the warnings are given in the first place.

**To Reproduce**

The following code suffices. Note that this is a slimmed down version of the actual component. I am aware of the new `<I18n>` component of 2.5.0.

```js
import React from 'react';
import PropTypes from 'prop-types';
import {withI18n} from "@lingui/react";

const HeadTitle = (props) => {
  return (
    <div>appName - {props.i18n._(props.title)}</div>
  )
};

export default withI18n()(HeadTitle)

HeadTitle.propTypes = {
  i18n: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
};
```

Note, if `withI18n` receives something "wrapped" (for lack of a better term), no warnings are given. For example, the following code works fine (that is, no warnings).

```
export default withI18n()(withCookies(HeadTitle))
```

([withCookies](https://www.npmjs.com/package/react-cookie) is used for an example, but anything will do, e.g. redux).

**Expected behavior**
No warnings are logged ;-)

**Additional context**

I suspect the change to `withI18n.js` in [this commit](https://github.com/lingui/js-lingui/commit/3804fa8b984b9b3d6c6c411adc3fe859205cec9a) is the culprit. But this is just a hunch.

**Comment by tricoder42** [comment]
Hey @Kernle32DLL,
this is indeed a bug introduced by `I18n` render prop component.

I tried to figure out quickly what's the issue but no luck. I'll take a look again later.

**Comment by tricoder42** [comment]
This was fixed in #302 and will be released soon.

**Comment by tricoder42** [comment]
Released in [v2.6.0](https://github.com/lingui/js-lingui/releases/tag/v2.6.0)
